hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/OzoneManagerStateMachine.java

```diff
@@ -166,6 +167,29 @@ public void notifyTermIndexUpdated(long currentTerm, long index) {
     computeAndUpdateLastAppliedIndex(index, currentTerm, null, false);
   }
 
+  /**
+   * Called to notify state machine about configuration changes.
+   * Configurations changes include addition of newly bootstrapped OM.
+   */
+  @Override
+  public void notifyConfigurationChanged(long term, long index,
+      RaftProtos.RaftConfigurationProto newRaftConfiguration) {
+    List<RaftProtos.RaftPeerProto> newPeers =
+        newRaftConfiguration.getPeersList();
```

Comment:
Looks like we need to call computeAndUpdateLastAppliedIndex. As applyLogToStateMachine in StateMachineUpdater when configEntry, we call 
      stateMachine.event().notifyConfigurationChanged(next.getTerm(), next.getIndex(), next.getConfigurationEntry());

As in OM StateMachine, we update lastAppliedIndex using computeAndUpdateLastAppliedIndex.

Reply:
Sorry reading more looks we should be fine, as for any no state machine log entry we call notifyTermIndexUpdated.

    if (!next.hasStateMachineLogEntry()) {
      stateMachine.event().notifyTermIndexUpdated(next.getTerm(), next.getIndex());
    }

Can you also confirm once?

Reply:
Correct. If the log entry is a configurationEntry or a metadataEntry, then RaftServerImpl calls notifyTermIndexUpdated on StateMachine.